pkg/distributor/validate_test.go

```diff
@@ -249,7 +299,7 @@ func TestValidateLabels(t *testing.T) {
 	require.NoError(t, testutil.GatherAndCompare(reg, strings.NewReader(`
 			# HELP cortex_discarded_samples_total The total number of samples that were discarded.
 			# TYPE cortex_discarded_samples_total counter
-			cortex_discarded_samples_total{group="custom label",reason="label_invalid",user="testUser"} 1
+			cortex_discarded_samples_total{group="custom label",reason="label_invalid",user="testUser"} 2
```

Comment:
nitpick to create this count out of the test cases instead of hardcoding it here. e.g. counting the number of expected errors. Would be amazing if you can do the same below; not sure how hard it would be

Reply:
I refactored the whole test to run each case for legacy and utf8 and compute the desired metrics using the test cases. This wasn't hard, but lengthy.